packages/app/src/app/pages/common/Modals/NewTeamModal/TeamMembers.tsx

```diff
@@ -1,25 +1,150 @@
-import React from 'react';
-import { Button, Stack, Textarea, Text } from '@codesandbox/components';
-import { useAppState } from 'app/overmind';
+import React, { FormEvent, useState } from 'react';
+import { Button, Stack, Text } from '@codesandbox/components';
+import { useAppState, useEffects } from 'app/overmind';
+import { StyledButton } from 'app/components/dashboard/Button';
+import { Textarea } from 'app/components/dashboard/Textarea';
+import { TeamMemberAuthorization } from 'app/graphql/types';
+
+function validateEmail(email: string) {
+  // Test for "anything@anything.anything" and check for
+  // multiple @ signs.
+  return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email);
+}
+
+function formatInvalidEmails(invalidEmails: string[]) {
+  const isSingleEmail = invalidEmails.length === 1;
+
+  if (isSingleEmail) {
+    return `${invalidEmails[0]}.`;
+  }
+
+  return invalidEmails.map((invalidEmail, index, arr) => {
+    const isFirstEmail = index === 0;
+    const isLastEmail = arr.length - 1 === index;
+
+    return `${
+      isLastEmail
+        ? ` and ${invalidEmail}.`
+        : `${isFirstEmail ? '' : ', '}${invalidEmail}`
+    }`;
+  });
+}
 
 export const TeamMembers: React.FC<{ onComplete: () => void }> = ({
   onComplete,
 }) => {
   const { activeTeamInfo } = useAppState();
+  const { gql } = useEffects();
```

Comment:
this validation behavior can be extracted into a custom hook maybe. not sure if it's going to be reused later